### PR TITLE
Prevent registration to event with email from user

### LINKED ISSUE #1850: Registrants with "fake" (LDAP:xxxx id) avatars are a bit broken
They have an avatar associated, so there's no "direct payment link" shown for them. However, they also cannot log in with that fake avatar.

We shouldn't associate fake avatars based on searching providers at all. Instead, we could maybe create a pending avatar or simply not associate an avatar with a registration unless the user is logged in...

### PR DESCRIPTION
**Pull request by jacquesd**
- prevents a user to register to an event with the email of another user
- prevents a guest to register to an event with the email of a user
- fixes #1850

### CODE REVIEW

File: indico/MaKaC/webinterface/rh/registrationFormDisplay.py [hunk 1]
```diff
@@ -112,11 +112,20 @@ def _process(self):
...
+        # Check if the email matches an existing user
+        user = get_user_by_email(email)
+        avatar = user.as_avatar if user else None
+        if user:
+            if not session.user:
+                return redirect_to_login(
+                    reason=_('The provided email ({email}) is linked to an Indico account. Please log in with that '
+                             'account first to register to the event with that email.').format(email=email))
+            elif session.user != user:
+                flash(_('The provided email ({email}) is linked to another Indico account. Please log in using that '
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/rh/registrationFormDisplay.py]: the wording sounds a bit weird with with the two "that" in that message

File: indico/MaKaC/webinterface/rh/registrationFormDisplay.py [hunk 2]
```diff
@@ -129,22 +138,22 @@ def _process(self):
             if self._conf.hasRegistrantByEmail(email):
                 raise FormValuesError("There is already a user with the email \"%s\". Please choose another one" % email)
         else:
-            if user.isRegisteredInConf(self._conf) or self._conf.hasRegistrantByEmail(email):
+            if user.as_avatar.isRegisteredInConf(self._conf) or self._conf.hasRegistrantByEmail(email):
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/rh/registrationFormDisplay.py]: `avatar`

File: indico/MaKaC/webinterface/rh/registrationFormDisplay.py [hunk 3]
```diff
@@ -129,22 +138,22 @@ def _process(self):
...
 
-        if user is not None:
-            rp.setAvatar(user)
+        if avatar is not None:
+            rp.setAvatar(avatar)
 
         notify_registration_confirmation(self._conf, rp)
 
-        if canManageRegistration and user != self._getUser():
+        if canManageRegistration and avatar != self._getUser():
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/rh/registrationFormDisplay.py]: `user != session.user`

**Comment by ThiefMaster** [comment]
Should we perform the same checks when modifying your registration (as you can change the email at this point)?

**Comment by pferreir** [comment]
@ThiefMaster, yes, why not? I can't see any disadvantages. Not doing the check during modification would render the fix useless...

**Comment by jacquesd** [comment]
@pferreir, @ThiefMaster it's [done](https://github.com/jacquesd/indico/commit/cf9dc22cb3e88185074a53d3892098182e8fd4a6#diff-d8349857e0cf9a9501b769f9d65bfb7fR242).

**Comment by pferreir** [comment]
Merged in cd82b6c869102c9e3111c6791e98ba61c2bd2e94

**Pull request closed without merging**